### PR TITLE
Unused type parameter in `hmc`

### PR DESCRIPTION
**Pull request by mschauer**

### CODE REVIEW

**Comment by mschauer** [comment]
One could also get rid of the `U` but perhaps the intention was to constrain it to traces?

**Comment by marcoct** [comment]
Thanks @mschauer ! Yes, that field should be `trace::Trace` and there should be no type parameters.

File: src/inference/hmc.jl [hunk 1]
```diff
@@ -24,7 +24,7 @@ Neal, Radford M. (2011), "MCMC Using Hamiltonian Dynamics", Handbook of Markov C
 """
 function hmc(
         trace::U, selection::Selection; L=10, eps=0.1,
```
Comment (mschauer) [inline src/inference/hmc.jl]:
```suggestion
        trace::Trace, selection::Selection; L=10, eps=0.1,
```

File: src/inference/hmc.jl [hunk 2]
```diff
@@ -24,7 +24,7 @@ Neal, Radford M. (2011), "MCMC Using Hamiltonian Dynamics", Handbook of Markov C
...
-        check=false, observations=EmptyChoiceMap()) where {T,U}
+        check=false, observations=EmptyChoiceMap()) where {U}
```
Comment (mschauer) [inline src/inference/hmc.jl]:
```suggestion
        check=false, observations=EmptyChoiceMap())
```

**Review by mschauer — COMMENTED** [review]
Suggestions from review

**Pull request merged**